### PR TITLE
Add Keycloak testplans

### PR DESCRIPTION
**Pull request by leonleoff**
Closes Gamify-IT/issues#258

### CODE REVIEW

**Review by madebyTimo — APPROVED** [review]

File: dev-manuals/test-plans/keycloak/u-keycloak-01-standard-login.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+Version: V1.0, 25.09.2022 \
+Author: Leon Layer
+
+## Description
+
+Standard login with user email and password.
+
+## Precondition
+
+No user is logged in. If the user was logged in, he has logged out so that he is no longer logged in via the cookies in the browser.
```
Comment (delvh) [inline dev-manuals/test-plans/keycloak/u-keycloak-01-standard-login.md:13]: So far no complaints. The only thing I'm missing is a test that **a new user can register**.

File: dev-manuals/test-plans/keycloak/u-keycloak-01-standard-login.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+## Typical procedure
+
+1. Open the landing-page
+2. Go to login
+3. Fill in correct username or correct email and correct password 
+4. Choose sign in
+
+## Alternative procedures
+
```
Comment (delvh) [inline dev-manuals/test-plans/keycloak/u-keycloak-01-standard-login.md:27]:
```suggestion

3.1. An incorrect username, email, or password is entered - The user is not redirected and is notified that their input was incorrect
```
By doing that, you can delete test `02` (don't forget to adapt the linkages)

What you can then do is create the `register` test I mentioned above as the new `02`.

File: dev-manuals/test-plans/keycloak/u-keycloak-07-user-roles-consideration-lecturer-interface-no-access.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
+# Keycloak: user roles consideration lecturer interface no access (`u.keycloak-07`)
```
Comment (delvh) [inline dev-manuals/test-plans/keycloak/u-keycloak-07-user-roles-consideration-lecturer-interface-no-access.md:1]:
```suggestion
# Keycloak: lecturer interface not linked for non-lecturers (`u.keycloak-07`)
```
Don't forget to adapt the filename and linkages.

File: dev-manuals/test-plans/keycloak/u-keycloak-07-user-roles-consideration-lecturer-interface-no-access.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+Version: V1.0, 25.09.2022 \
+Author: Leon Layer
+
+## Description
+
+A user without a lecturer role, is logged in. His role causes the lecturer interface to be not displayed on the start page.
+
+## Precondition
+
+A user without a lecturer role is logged in.
```
Comment (delvh) [inline dev-manuals/test-plans/keycloak/u-keycloak-07-user-roles-consideration-lecturer-interface-no-access.md:13]:
```suggestion
The user stays logged in. 
```

File: dev-manuals/test-plans/keycloak/u-keycloak-07-user-roles-consideration-lecturer-interface-no-access.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
+# Keycloak: user roles consideration lecturer interface no access (`u.keycloak-07`)
+
+
+Version: V1.0, 25.09.2022 \
+Author: Leon Layer
+
+## Description
+
+A user without a lecturer role, is logged in. His role causes the lecturer interface to be not displayed on the start page.
```
Comment (delvh) [inline dev-manuals/test-plans/keycloak/u-keycloak-07-user-roles-consideration-lecturer-interface-no-access.md:9]:
```suggestion
A user who is not a lecturer is logged in. They will not see a link to the lecturer interface.
```

File: dev-manuals/test-plans/keycloak/u-keycloak-06-user-roles-consideration-lecturer-interface-access.md [hunk 6]
```diff
@@ -0,0 +1,34 @@
+# Keycloak: user roles consideration lecturer interface access (`u.keycloak-06`)
```
Comment (delvh) [inline dev-manuals/test-plans/keycloak/u-keycloak-06-user-roles-consideration-lecturer-interface-access.md:1]:
```suggestion
# Keycloak: only lecturers get a link to the lecturer interface (`u.keycloak-06`)
```

File: dev-manuals/test-plans/keycloak/u-keycloak-07-user-roles-consideration-lecturer-interface-no-access.md [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+## Precondition
+
+A user without a lecturer role is logged in. 
+
+## Postcondition
+
+The possibility to open the lecturer interface is not displayed to the user.
```
Comment (delvh) [inline dev-manuals/test-plans/keycloak/u-keycloak-07-user-roles-consideration-lecturer-interface-no-access.md:17]: Again, isn't that basically just an alternative procedure for `06`?

File: dev-manuals/test-plans/keycloak/u-keycloak-08-user-roles-consideration-change.md [hunk 8]
```diff
@@ -0,0 +1,37 @@
+# Keycloak: user roles consideration change (`u.keycloak-08`)
```
Comment (delvh) [inline dev-manuals/test-plans/keycloak/u-keycloak-08-user-roles-consideration-change.md:1]:
```suggestion
# Keycloak: user role change (`u.keycloak-08`)
```

File: dev-manuals/test-plans/keycloak/u-keycloak-08-user-roles-consideration-change.md [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+
+
+Version: V1.0, 25.09.2022 \
+Author: Leon Layer
+
+## Description
+
+If a user has been assigned the role of lecturer, that means that he can access the configurations of the overworld. After logging in, this user can call up the lecture interface on his start page to edit the course. This means that the lecturer interface is not displayed to a user without a lecturer role. If you change the role of this user to a lecturer, the lecturer interface is displayed.
```
Comment (delvh) [inline dev-manuals/test-plans/keycloak/u-keycloak-08-user-roles-consideration-change.md:9]:
```suggestion
A user who was previously not a lecturer is allowed to do all lecturer-only things once he is promoted to `lecturer`.  
```

File: dev-manuals/test-plans/keycloak/u-keycloak-08-user-roles-consideration-change.md [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+
+## Precondition
+
+A user without lecturer status is logged in. The user is not offered the option to open the lecturer interface on the start page.
```
Comment (delvh) [inline dev-manuals/test-plans/keycloak/u-keycloak-08-user-roles-consideration-change.md:13]:
```suggestion
A user without lecturer status is logged in. The user is on the start page. The user is not offered the option to open the lecturer interface.
```

File: dev-manuals/test-plans/keycloak/u-keycloak-08-user-roles-consideration-change.md [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+
+## Postcondition
+
+The user sees the lecturer interface.
+
+## Typical procedure
+
+1. The role of the logged-in user is changed in the Keylok admin console.
```
Comment (delvh) [inline dev-manuals/test-plans/keycloak/u-keycloak-08-user-roles-consideration-change.md:21]:
```suggestion
1. The user gets the role `lecturer` through the Keyloak admin console.
```

File: dev-manuals/test-plans/keycloak/u-keycloak-08-user-roles-consideration-change.md [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+2. The logged in user waits until the role change is processed by the server.
+3. The user reloads the home page
```
Comment (delvh) [inline dev-manuals/test-plans/keycloak/u-keycloak-08-user-roles-consideration-change.md:23]:
```suggestion
3. The user reloads the page
```

**Review by robojones — APPROVED** [review]

**Review by Gr33ndev — APPROVED** [review]

**Pull request merged**